runtime/domain.c

```diff
@@ -84,7 +88,12 @@ struct dom_internal {
   struct domain state;
   struct interruptor interruptor;
 
-  caml_plat_mutex roots_lock;
+  /* backup thread */
+  int backup_thread_running;
```

Comment:
I couldn't see where we initialize this variable to a known state.
I see `backup_thread_msg` getting done in `create_domain`
and the lock/cond pair in `caml_init_domains`

Reply:
Done.